caldav/server.go

```diff
@@ -42,6 +40,11 @@ type Backend interface {
 	webdav.UserPrincipalBackend
 }
 
+type MultiCalendarBackend interface {
+	Calendars(ctx context.Context) ([]*Calendar, error)
```

Comment:
I'm also not sure if it makes sense to use `[]*Calendar` here, or if we could just go with `[]Calendar`? I went with this now, since it was already used in `propFindAllCalendars`